plugins/node/opentelemetry-instrumentation-aws-sdk/package.json

```diff
@@ -49,7 +49,7 @@
     "@opentelemetry/core": "^1.0.0",
     "@opentelemetry/instrumentation": "^0.28.0",
     "@opentelemetry/semantic-conventions": "^1.0.0",
-    "@opentelemetry/propagation-utils": "^0.27.0"
+    "@opentelemetry/propagation-utils": "0.27.0"
```

Comment:
pin dependencies within the same learna monorep, forgot this one in a previous PR #984

Reply:
Please help me understand why are we supposed to pin "monorepo-local" packages / why should we make those follow an different rule than external pacakges.

There are more:

- `@opentelemetry/propagator-aws-xray@^1.0.1` in `@opentelemetry/instrumentation-aws-lambda`
- `@opentelemetry/instrumentation-express@^0.28.0` in `express-example`
- `@opentelemetry/instrumentation-document-load@^0.27.1` in `@opentelemetry/browser-extension-autoinjection`
- all instrumentations in `@opentelemetry/auto-instrumentations-*`

Are these also supposed to be pinned? Why not?

Reply:
We do it in core repo and I see no reason to do it different here. They are released in one shot and user should update all of them or nothing. So it's always consistent as during tests here.
At least this is my understanding based on the setup in core repo.

Whenever I had a mixup of versions it ended up in a nightmare.

And yes, you are right, there are a few dependencies I haven't seen yet. In general I have the feeling that dependency management in this (and core) repo should be more automated but till now I found no time to get familiar with renovate bot.

Reply:
I'm still not sure what is the best way to setup the dependencies. It seems that every release made has it's own followup problems regarding dependencies. Not sure if we are able to get to a setup which just works.

Reply:
> We do it in core repo and I see no reason to do it different here. They are released in one shot and user should update all of them or nothing. So it's always consistent as during tests here.

I'm all for consistency, just trying to understand the reasoning behind it. So the main reason is to mimic the repository setup since they are linked in CI. That makes sense.

> And yes, you are right, there are a few dependencies I haven't seen yet. In general I have the feeling that dependency management in this (and core) repo should be more automated but till now I found no time to get familiar with renovate bot.

I'm also totally for automation, but currently I personally don't even always understand the rules to automate them. Hence me bombarding you with questions. :)

Reply:
By "examples I listed above" I meant:

> There are more:
>
> - @opentelemetry/propagator-aws-xray@^1.0.1 in @opentelemetry/instrumentation-aws-lambda
> - @opentelemetry/instrumentation-express@^0.28.0 in express-example
> - @opentelemetry/instrumentation-document-load@^0.27.1 in @opentelemetry/browser-extension-autoinjection
> - all instrumentations in @opentelemetry/auto-instrumentations-*

Which by the rule of "monorepo local deps should be pinned" are currently inconsistent.

What comes to `@opentelemetry/instrumentation` package, I'd leave that to `^0.27` for now **and** specify http and grpc instrumentations accordingly to a version that uses `^0.27` - we can then update that separately. What do you think?

Reply:
> What comes to @opentelemetry/instrumentation package, I'd leave that to ^0.27 for now and specify http and grpc instrumentations accordingly to a version that uses ^0.27 - we can then update that separately. What do you think?

If we depend on old http instrumentation this will pull in old SDK components and likely to new problems.